### PR TITLE
#2363 Spike for conditional form fields

### PR DESCRIPTION
**Pull request by kevinhowbrook**
Do not merge

**Based off https://github.com/torchbox/rca-wagtail-2019/pull/817**

Spike ticket [#2636](https://projects.torchbox.com/projects/rca-django-cms-project/tickets/2363)

This pull request is a spike to serve as proof we can implement conditional / dependant form fields. The Programme dropdown on the form should change the list of scholarships offered:

image

### CODE REVIEW

File: rca/project_styleguide/templates/patterns/pages/scholarships/scholarship_enquiry.html [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        var url = $("#scholarship-form").attr("data-scholarships-url");
+
+        let searchParams = new URLSearchParams(window.location.search)
+        var programmeId = searchParams.get('programme');
+        if (programmeId){
+            console.log('INITIAL LOAD OF SCHOLARSHIPS - ADJUSTING AS PROGRAMME FOUND')
+            $('#id_programme').val(programmeId).change();
+        }
+    });
+</script>
```
Comment (kevinhowbrook) [inline rca/project_styleguide/templates/patterns/pages/scholarships/scholarship_enquiry.html:128]: Task for @nicklee to implement this better

File: rca/scholarships/views.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+from rca.scholarships.models import ScholarshipSnippet
+
+from .forms import ScholarshipForm
+
+logger = logging.getLogger(__name__)
+
+
+def load_scholarships(request):
+    programme = request.GET.get("programme")
```
Comment (richbrennan) [inline rca/scholarships/views.py:16]: Suggest we pass the programme in the url: `ajax/load-scholarships/1/`

File: rca/scholarships/views.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+from django.views.generic import FormView
+
+from rca.scholarships.models import ScholarshipSnippet
+
+from .forms import ScholarshipForm
+
+logger = logging.getLogger(__name__)
+
+
+def load_scholarships(request):
```
Comment (richbrennan) [inline rca/scholarships/views.py:15]: Suggest make AJAX-only: https://docs.djangoproject.com/en/3.2/ref/request-response/#django.http.HttpRequest.accepts

**Review by richbrennan — COMMENTED** [review]
Approach looks good

**Pull request closed without merging**